File: backend/app/schemas/ticket.py
from pydantic import BaseModel, Field, ConfigDict
from typing import Optional, List
from datetime import datetime

class TagBase(BaseModel):
    id: int
    name: str
    color: str
    
    model_config = ConfigDict(from_attributes=True)

class TicketBase(BaseModel):
    title: str = Field(..., min_length=1, max_length=200)
    description: Optional[str] = None

class TicketCreate(TicketBase):
    tag_ids: Optional[List[int]] = []

class TicketUpdate(BaseModel):
    title: Optional[str] = Field(None, min_length=1, max_length=200)
    description: Optional[str] = None
    tag_ids: Optional[List[int]] = None

class TicketResponse(TicketBase):
    id: int
    is_completed: bool
    created_at: datetime
    updated_at: datetime
    tags: List[TagBase] = []
    
    model_config = ConfigDict(from_attributes=True)

class TicketListResponse(BaseModel):
    tickets: List[TicketResponse]

File: frontend/src/types/ticket.ts
import type { Tag } from './tag';

export interface Ticket {
  id: number;
  title: string;
  description: string | null;
  is_completed: boolean;
  created_at: string;
  updated_at: string;
  tags?: Tag[];
}

export interface CreateTicketRequest {
  title: string;
  description?: string;
  tag_ids?: number[];
}

export interface UpdateTicketRequest {
  title?: string;
  description?: string;
  tag_ids?: number[];
}

export interface TicketListResponse {
  tickets: Ticket[];
}
